kyo-core/shared/src/main/scala/kyo/tags.scala

```diff
@@ -1,41 +1,206 @@
 package kyo
 
-import izumi.reflect.dottyreflection.TypeInspections
-import izumi.reflect.macrortti.LightTypeTag
+import scala.annotation.tailrec
 import scala.quoted.*
 
-opaque type Tag[T] = String
+case class Tag[T](tpe: String) extends AnyVal
 
 object Tag:
 
-    given canEqual[T, U]: CanEqual[Tag[T], Tag[U]] = CanEqual.derived
+    import internal.*
+
+    inline given apply[T >: Nothing]: Tag[T] = ${ tagImpl[T] }
```

Comment:
Good question! I think we actually shouldn't allow `Tag[Nothing]`. In Kyo's usage of tags, it's essentially an inconsistent state since tags are meant to represent concrete types, and `Nothing`, although it has a class symbol, has no runtime representation. I've changed the macro to fail for `Tag[Nothing]` and added a test for it. It caught an inssue in `streamsTest`! I've removed the type constraint since it isn't strict, and the macro will detect it anyway.